src/rust/lib/graph-editor/src/component/visualization/class.rs

```diff
@@ -20,11 +23,31 @@ pub struct EnsoCode {
 }
 
 /// Type alias for a string representing an enso type.
-#[derive(Clone,CloneRef,Debug)]
+#[derive(Clone,CloneRef,Debug,PartialEq,Eq,Hash)]
 pub struct EnsoType {
     content: Rc<String>
 }
 
+impl From<String> for EnsoType {
+    fn from(source:String) -> Self {
+        EnsoType { content:Rc::new(source) }
+    }
+}
+
+impl From<&str> for EnsoType {
+    fn from(source:&str) -> Self {
+        EnsoType { content:Rc::new(source.to_string()) }
+    }
+}
+
+/// Contains general information about a visualization.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub struct ClassAttributes {
+    pub name        : String,
+    pub input_types : Vec<EnsoType>,
+}
```

Comment:
Im still tending to name it `visualization::class::Signature` instead. (btw you are in `visualization::class` so no need to repeat word `Class` in the name)